org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGenerator.xtend

```diff
@@ -338,7 +338,7 @@ class XtextGenerator extends AbstractWorkflowComponent2 {
 			if (root.isFile(pluginXml.path)) { 
 				// only write plugin.xml_gen if entries exist and content differs
 				if (!pluginXml.entries.isEmpty
-					&& root.readTextFile(pluginXml.path) != pluginXml.getContent 
+					&& root.readTextFile(pluginXml.path)?.toString != pluginXml.getContent?.toString
```

Comment:
Should we introduce TextFileAccess#getContentString to easier avoid the same confusion / bug pattern in the future?